docs/docs/30-administration/40-encryption.md

```diff
@@ -0,0 +1,64 @@
+# Secrets encryption
+
+By default, Woodpecker does not encrypt secrets in its database. You can enable encryption 
+using simple AES key or more advanced [Google TINK](https://developers.google.com/tink) encryption.
+
+## Common
+
+### Enabling secrets encryption
+
+To enable secrets encryption and encrypt all existing secrets in database set
+`WOODPECKER_ENCRYPTION_KEY`, `WOODPECKER_ENCRYPTION_KEY_FILE` or `WOODPECKER_ENCRYPTION_TINK_KEYSET_PATH` environment 
+variable depending on encryption method of your choice.
+
+After encryption is enabled you will be unable to start Woodpecker server without providing valid encryption key!
+
+### Disabling encryption and decrypting all secrets
+
+To disable secrets encryption and decrypt database you need to start server with valid
+`WOODPECKER_ENCRYPTION_KEY` or `WOODPECKER_ENCRYPTION_TINK_KEYSET_FILE` environment variable set depending on
+enabled encryption method, and `WOODPECKER_ENCRYPTION_DISABLE` set to true.
+
+After secrets was decrypted server will proceed working in unencrypted mode. You will not need to use "disable encryption"
+variable or encryption keys to start server anymore.
+
+
+## AES
+Simple AES encryption.
```

Comment:
AES **GCM** ...

why did you choose GCM over CBC @antomy-gc ?

Reply:
https://thehftguy.com/2020/04/20/what-aes-ciphers-to-use-between-cbc-gcm-ccm-chacha-poly/
>AES-GCM (Galois Counter Mode)
The most widely used block cipher worldwide.
Mandatory as of TLS 1.2 (2008) and used by default by most clients.
RFC 5288 year 2008 https://tools.ietf.org/html/rfc5288AES-GCM (Galois Counter Mode)

> AES-CBC
First historic block cipher for AES.
CBC mode is insecure and must not be used. It’s been progressively deprecated and removed from SSL libraries.
Introduced with TLS 1.0 year 2002. Superseded by GCM in TLS 1.2 year 2008. Removed in TLS 1.3 year 2018.
RFC 3268 year 2002 https://tools.ietf.org/html/rfc3268

Google also recommends AEAD with GCM key as suitable and secure enough for most uses.
https://developers.google.com/tink/aead

Reply:
If I remember correctly Hashicorp Vault also uses AES GCM

Reply:
can i resolve it and merge?

Reply:
@6543 can we resolve it and merge?